controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -96,6 +96,18 @@ func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 
 	expectedSnapshot, err := a.prepareSnapshotForPipelineRun(a.pipelineRun, a.component, a.application)
 	if err != nil {
+		// If PipelineRun result returns cusomized error stop reconciliation and update PLR annotation
+		if h.IsMissingInfoInPipelineRunError(err) {
+			// update the build PLR annotation with the error cusomized Reason and Value
+			err = a.annotateBuildPipelineRunWithCreateSnapshotAnnotation(err)
+			if err != nil {
+				a.logger.Error(err, "Could not add create snapshot annotation to build pipelineRun", h.CreateSnapshotAnnotationName, a.pipelineRun)
+			}
+			a.logger.Error(err, "Build PipelineRun %s failed with error, should be fixed and re-run manually", a.pipelineRun)
+			return controller.ContinueProcessing()
```

Comment:
comment says that it should stop reconciliation, maybe update comment to not say stop reconciliation

Reply:
don't we need to remove finalizer here as well?

Reply:
fixed the comment , 
nice catch for the finalizer , I've replaced   `return controller.ContinueProcessing()`  with  `removeFinalizerAndContinueProcessing()`